src/common/snippets/src/op/reduce.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/op/reduce.hpp"
+
+#include "snippets/itt.hpp"
+#include "snippets/lowered/port_descriptor.hpp"
+
+namespace ov {
+namespace snippets {
+namespace op {
+namespace {
+void compute_and_set_reduce_subtensors(const std::shared_ptr<ReduceBase>& reduce) {
+    OPENVINO_ASSERT(reduce->get_input_partial_shape(0).rank().is_static(),
+                    "Subtensors can be automatically calculated only for reduce with static rank.");
+    const auto reduce_rank = reduce->get_input_partial_shape(0).size();
+    const auto axis = reduce->get_axis();
+
+    std::vector<size_t> subtensor(reduce_rank, 1);
```

Comment:
Subtensors currently are 1D for standard snippets ops or 2D for TPP.
Are you sure it's a good idea to make subtensor the same size as the shape?
I mean, it's a step in a right direction, but wouldn't it be better if change it everywhere in a separate activity?

My concern is that some operations have 1D subternsor, some 2D - which is bad enough already, now some operations will have subtensors of unknown rank (could be different for different shapes).

Reply:
I moved subtensor composing behavior from `SetSoftmaxPorts`: so we already had `subtensor_rank = shape_rank` cases. I'd rather keep the old logic